### DISCUSSION TITLE
helping new contributors understand VCR usage

### DISCUSSION THREAD

**Discussion by tansaku** [vcr/vcr#861] [Ideas]
I've opened a thread on the mailing list, but there was mention there that GitHub issues were sometimes preferred, so here's a link to my questions there about helping new contributors how to understand how VCR is being used in a project:

https://groups.google.com/forum/#!topic/vcr-ruby/8DoA7MUq-38

I also have an idea that maybe it would be useful to have something like dependabot, but for VCR cached files, i.e. a bot that would ensure that individual VCR cache files were deleted occasionally and a pull request submitted with the results.  That way one could avoid the pile up of old cache files, in the same way that one can hopefully avoid lots of old gem dependencies with dependabot ...

**Comment by roschaefer** [comment]
Consider that it could be difficult to re-record cached VCR cassettes automatically. E.g. I have an app in which only core maintainers have access to private API keys. VCR helps to filter those keys from the cassettes but only core maintainers will be able to re-record those cassettes. For this scenario, I guess, the only way is to re-record the cassettes manually, once in a while.

**Comment by tansaku** [comment]
that's a good point @roschaefer - although one could have a system like we have for CI where the API keys are stored encrypted with the bot that then sends the submissions.  That does make the system more complex of course and require greater trust ...

**Comment by tansaku** [comment]
I was just thinking about this some more, i.e. what the steps would be to make it work.  I think they are:

VCRbot
---------

1) identify directory for VCR files
2) choose a file at random (that's over some setting old)
3) delete that file and re-record
4) commit to git and submit to CI?

**Comment by krainboltgreene** [comment]
Honestly, I'm of two minds:

  1. vcr is a last resort for test mocks, we should really be pushing for mocks at a higher level in the stack than the network
  2. The proposed idea is kinda like mutation specs, which is cool.

**Comment by krainboltgreene** [comment]
Fair warning though, that might get your CI instance banned from some APIs haha.
